lib/doctor.ts

```diff
@@ -28,150 +28,207 @@ export class Doctor implements NativeScriptDoctor.IDoctor {
 		return false;
 	}
 
-	public async getWarnings(config?: NativeScriptDoctor.ISysInfoConfig): Promise<NativeScriptDoctor.IWarning[]> {
-		let result: NativeScriptDoctor.IWarning[] = [];
+	public async getInfos(config?: NativeScriptDoctor.ISysInfoConfig): Promise<NativeScriptDoctor.IInfo[]> {
+		let result: NativeScriptDoctor.IInfo[] = [];
 		const sysInfoData = await this.sysInfo.getSysInfo(config);
 
-		const androidHomeValidationErrors = this.androidToolsInfo.validateAndroidHomeEnvVariable();
-		if (androidHomeValidationErrors.length > 0) {
-			result = result.concat(androidHomeValidationErrors);
-		}
-
-		if (!sysInfoData.adbVer) {
-			result.push({
-				warning: "WARNING: adb from the Android SDK is not installed or is not configured properly. ",
-				additionalInformation: "For Android-related operations, the NativeScript CLI will use a built-in version of adb." + EOL
-				+ "To avoid possible issues with the native Android emulator, Genymotion or connected" + EOL
-				+ "Android devices, verify that you have installed the latest Android SDK and" + EOL
-				+ "its dependencies as described in http://developer.android.com/sdk/index.html#Requirements" + EOL,
+		result = result.concat(
+			this.processValidationErrors({
+				warnings: this.androidToolsInfo.validateAndroidHomeEnvVariable(),
+				infoMessage: "Your ANDROID_HOME environment variable is set and points to correct directory.",
 				platforms: [Constants.ANDROID_PLATFORM_NAME]
-			});
-		}
-
-		if (!sysInfoData.isAndroidSdkConfiguredCorrectly) {
-			result.push({
-				warning: "WARNING: The Android SDK is not installed or is not configured properly.",
-				additionalInformation: "You will not be able to run your apps in the native emulator. To be able to run apps" + EOL
-				+ "in the native Android emulator, verify that you have installed the latest Android SDK " + EOL
-				+ "and its dependencies as described in http://developer.android.com/sdk/index.html#Requirements" + EOL,
+			}), [
```

Comment:
You don't actually need an array as a second argument to `result.concat`
```JavaScript
> [].concat([1,2,3], 4, 5, [6, 7, 8], "asd", {})
[ 1, 2, 3, 4, 5, 6, 7, 8, 'asd', {} ]
```

I'd prefer we do not add additional brackets where not necessary